desktop-src/direct3dhlsl_current/hlsl-built-ins-functions_math_logic.md

```diff
@@ -0,0 +1,3332 @@
+# Math and Logic Operations
+
+### abs
+
+Returns the absolute value of the specified value.
+
+
+
+| ret abs(*x*) |
+|--------------|
+
+
+
+ 
+
+#### Parameters
+
+
+
+| Item                                                   | Description                            |
+|--------------------------------------------------------|----------------------------------------|
+| <span id="x"></span><span id="X"></span>*x*<br/> | \[in\] The specified value.<br/> |
+
+
+
+ 
+
+#### Return Value
+
+The absolute value of the *x* parameter.
+
+#### Type Description
+
+
+
+| Name  | [**Template Type**](dx-graphics-hlsl-intrinsic-functions.md)                                                  | [**Component Type**](dx-graphics-hlsl-intrinsic-functions.md)                 | Size                           |
+|-------|----------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------|--------------------------------|
+| *x*   | [**scalar**](dx-graphics-hlsl-intrinsic-functions.md), **vector**, or **matrix** | [**float**](/windows/desktop/WinProg/windows-data-types), [**int**](/windows/desktop/WinProg/windows-data-types) | any                            |
```

Comment:
If types are linked, they should probably link to the appropriate HLSL type definitions, not windows-data-types, which doesn't even define these types, and instead defines a bunch of types from Windows headers that are not applicable to HLSL at all.  For now, I'd say just get rid of the links here.

Reply:
Alright, fixed links to the proper locations in the current documentation.